sop-website/docs/Waveform-Data/Waveform-Extension.mdx

```diff
@@ -0,0 +1,222 @@
+|:-----:|:--------------------------------:|:---------:|:----------------------:|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------:|
+| 1     | waveform_feature_id              | int       | Yes                    | Generate a unique surrogate key (e.g., via sequence or UUID). Each derived feature must have its own ID.                                                                                                                                                                                                                                                                       |
+| 2     | waveform_occurrence_id           | int       | Yes                    | Foreign key to waveform_occurrence. Must be assigned from the session that provided the raw waveform data. Extracted from upstream linkage or stored in intermediate metadata pipeline.                                                                                                                                                                                        |
+| 3     | waveform_registry_id             | int       | Yes                    | Foreign key to waveform_registry. Identifies the specific file from which the feature was extracted. Ensure file was processed and exists in the registry.                                                                                                                                                                                                                     |
+| 4     | waveform_channel_metadata_id     | int       | Yes                    | Foreign key to waveform_channel_metadata. Indicates the exact channel used to compute the feature (e.g., ECG Lead II). Use the signal name and channel index to link.                                                                                                                                                                                                          |
+| 5     | measurement_id / observation_id  | int       | Conditionally Required | If the derived feature matches an existing OMOP MEASUREMENT (e.g., heart rate) or OBSERVATION (e.g., “apnea event”), populate the appropriate foreign key. Use LOINC/OMOP vocabularies. Leave null if no standard concept applies.                                                                                                                                             |
```

Comment:
Should we have separate fields for `measurement_id` and `observation_id`? I wonder if this could be helpful when querying this table. For example, if a researcher was interested in whether there was a related `measurement_id` but didn't care if there is a related `observation_id`. I'm sure they could handle that with a little extra coding but figured I'd raise this since the thought crossed my mind.